dsl/pace/dsl/dace/utils.py

```diff
@@ -20,3 +24,97 @@ def __enter__(self):
     def __exit__(self, _type, _val, _traceback):
         elapsed = time.time() - self.start
         DaCeProgress.log(self.prefix, f"{self.label}...{elapsed}s.")
+
+
+def NaN_checker(sdfg: dace.SDFG):
```

Comment:
Nit. [ref](https://developer.rhino3d.com/guides/rhinopython/python-code-conventions/#:~:text=Names%20and%20Capitalization,-In%20Python%20names&text=Python%20does%20not%20allow%20characters,will%20be%20all%20lower%20case.)
```suggestion
def sdfg_nan_checker(sdfg: dace.SDFG):
```

This will require refactor changes.